### PR TITLE
Attempt a cache

### PR DESCRIPTION
**Pull request by Hoverbear**
Experimenting off #34 with a build cache for linux

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,46 +8,61 @@ on:
...
       - uses: cachix/install-nix-action@v17
         with:
           extra_nix_config: |
             access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      - name: Cache lint store (x86_64-linux)
+        id: build-store-x86_64-linux
+        uses: actions/cache@v3
+        with:
+          path: ~/.ci-store
+          key: lint-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/Flake.lock') }}-v1
```
Comment (cole-h) [inline .github/workflows/ci.yml:23]: Dunno if `hashFiles` is case-sensitive but I'd bet it is:

```suggestion
          key: lint-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/flake.lock') }}-v1
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,46 +8,61 @@ on:
...
         with:
           extra_nix_config: |
             access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
-      # Runs clippy as part of the preBuild.
+      - name: Cache build store (x86_64-linux)
+        id: build-store-x86_64-linux
+        uses: actions/cache@v3
+        with:
+          path: ~/.ci-store
+          key: build-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/Flake.lock') }}-v1
```
Comment (cole-h) [inline .github/workflows/ci.yml:47]:
```suggestion
          key: build-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/flake.lock') }}-v1
```

**Review by cole-h — COMMENTED** [review]
I wonder if we wouldn't have to do all this hacky stuff if we saved `/nix/store` after everything, and then restored it before installing Nix?

**Comment by Hoverbear** [comment]
I tried caching `/nix` but the cache action doesn't run as root :(

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -8,46 +8,61 @@ on:
...
       - uses: cachix/install-nix-action@v17
         with:
           extra_nix_config: |
             access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      - name: Cache lint store (x86_64-linux)
+        id: build-store-x86_64-linux
+        uses: actions/cache@v3
+        with:
+          path: ~/.ci-store
+          key: lint-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/flake.lock') }}-v1
```
Comment (cole-h) [inline .github/workflows/ci.yml:23]: Whoops, also missed these.

```suggestion
        id: lint-store-x86_64-linux
        uses: actions/cache@v3
        with:
          path: ~/.ci-store
          key: lint-store-x86_64-linux-${{ hashFiles('**/Cargo.lock', '**/flake.lock') }}-v1
```

**Review by cole-h — APPROVED** [review]
After my below comment, LGTM.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Hoverbear) [reply .github/workflows/ci.yml:23]: Oh!!!!

**Pull request merged**